src/baldr/graphreader.cc

```diff
@@ -37,31 +37,61 @@ tile_gone_error_t::tile_gone_error_t(std::string prefix, baldr::GraphId edgeid)
 
 GraphReader::tile_extract_t::tile_extract_t(const boost::property_tree::ptree& pt) {
   // if you really meant to load it
-  if (pt.get_optional<std::string>("tile_extract")) {
+  auto extract_fp = pt.get_optional<std::string>("tile_extract");
+  if (extract_fp) {
     try {
-      // load the tar
-      archive.reset(new midgard::tar(pt.get<std::string>("tile_extract")));
```

Comment:
yeah i would just keep this exactly the way it is already and instead of doing work here, i would go into midgard::tar and i would add a new feature to that interface that allows us to inject functionality for parsing the contents from an index file. let me hack up a proposal somehow...

Reply:
yeah much nicer..